recipes/7bitconf/all/conanfile.py

```diff
@@ -0,0 +1,144 @@
+        compiler = self.settings.compiler
+        compiler_name = str(compiler)
+
+        if compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+        minimum_version = self._minimum_compilers_version.get(compiler_name, False)
+        if minimum_version and Version(compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"Requires compiler {compiler_name} minimum version: {minimum_version} with C++17 support."
+            )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        if not self.options.header_only:
+            tc = CMakeToolchain(self)
+            tc.variables["_7BIT_CONF_BUILD_EXAMPLES"] = False
+            tc.variables["_7BIT_CONF_BUILD_TESTS"] = False
+            tc.variables["_7BIT_CONF_BUILD_SINGLE_HEADER"] = False
+            tc.variables["_7BIT_CONF_INSTALL"] = True
+            tc.variables["_7BIT_CONF_LIBRARY_TYPE"] = self.getSevenBitConfLibraryType()
+            tc.generate()
+        cmake_deps = CMakeDeps(self)
+        cmake_deps.generate()
+
+    def getSevenBitConfLibraryType(self):
+        if self.options.header_only:
+            return "HeaderOnly"
+        elif self.options.shared:
+            return "Shared"
+        else:
+            return "Static"
+
+    def build(self):
+        if not self.options.header_only:
+            cmake = CMake(self)
+            cmake.configure()
+            cmake.build()
+
+    def package(self):
+        copy(
+            self,
+            "LICENSE",
+            dst=os.path.join(self.package_folder, "licenses"),
+            src=self.source_folder,
+        )
+        if self.options.header_only:
+            copy(
+                self,
+                src=os.path.join(self.source_folder, "Include"),
+                pattern="*.hpp",
+                dst=os.path.join(self.package_folder, "include"),
+            )
+        else:
+            cmake = CMake(self)
+            cmake.install()
+            rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "7bitConf")
+        self.cpp_info.set_property("cmake_target_name", "7bitConf::7bitConf")
+        self.cpp_info.requires = ["taocpp-json::taocpp-json"]
+
+        if self.options.header_only:
+            self.cpp_info.libs = []
+            self.cpp_info.bindirs = []
+        else:
+            suffix = "d" if self.settings.build_type == "Debug" else ""
+            self.cpp_info.libs = ["7bitConf" + suffix]
+
+            if self.settings.os in ["Linux", "FreeBSD"]:
+                self.cpp_info.system_libs.append("m")
+
+        # TODO: to remove in conan v2 once cmake_find_package_* generators removed
+        self.cpp_info.names["cmake_find_package"] = "7bitConf"
+        self.cpp_info.names["cmake_find_package_multi"] = "7bitConf"
```

Comment:
This is no longer necessary for new recipes, you can drop these lines

Reply:
Thank you, I have removed these lines.